lib/network.js

```diff
@@ -13,7 +13,9 @@ module.exports = function (archive, opts, cb) {
     stream: opts.stream
   }, opts)
   var swarm = disc(swarmDefaults(swarmOpts))
-  swarm.once('error', function () {
+  swarm.once('error', function (err) {
+    // Don't magically use any port if the users has requested a specific port
+    swarm.emit('errorport', err)
     swarm.listen(0)
```

Comment:
Should we call this if `opts.port` isn't set?

Reply:
I originally added an if for that too - but the errorport handling shortcircuits before that, so it seemed cleaner to not add a duplicate if